### PR TITLE
Fix overwrite of httpd config during update

### LINKED ISSUE #16: Changed httpd configuration will be overwritten in update
Due to changes in how CentOS 8 automatically creates self-signed certificates during start of the httpd service, the default httpd configuration file for privacyIDEA will be overwritten on each update.

### PR DESCRIPTION
**Pull request by plettich**
- The configuration file is marked as "noreplace" which should always
  keep the old/modified version
- Fix a broken encoding in a Python file on CentOS 7
- Hide error message during database check

Fixes #16

### CODE REVIEW

File: SPECS/privacyidea-server.spec [hunk 1]
```diff
@@ -125,16 +125,13 @@ fi
 
 ###################################################
 # The webserver
-# first we need to start the webserver to let it create the self-signed certificates
-systemctl start httpd
-# mkdir -p /var/run/wsgi
+# first we need to create the self-signed certificates on CentOS 8.
+[[ -x /usr/libexec/httpd-ssl-gencerts ]] && /usr/libexec/httpd-ssl-gencerts
```
Comment (cornelinux) [inline SPECS/privacyidea-server.spec:129]: What would happen during an update, if the certificate files have been manually updated with productive certificates, while the privacyidea.conf has not been touched in the setup.
Would the certificate files be overwritten by this?

**Review by cornelinux — COMMENTED** [review]
Please check my remark

File: SPECS/privacyidea-server.spec (hunk 1 above)
Comment (pablo-knight) [reply SPECS/privacyidea-server.spec:129]: Replayed this senario with a untouched privacyidea.conf, and a modifyed certificate. Certificate was not changed.
Second senario with changes in privacyidea.conf and a other certificate. Both was not overwritten.  Pi-Update create a new  file: privacyidea.conf.rpmnew how assume.

Comment (pablo-knight) [reply SPECS/privacyidea-server.spec:129]: Also tested in centos8 with the same result as above.

Comment (cornelinux) [reply SPECS/privacyidea-server.spec:129]: @pablo-knight  Do you merge it?

Comment (pablo-knight) [reply SPECS/privacyidea-server.spec:129]: merged

**Pull request merged**